incubator/exporter-otlp/src/test/java/io/aklivity/zilla/runtime/exporter/otlp/internal/serializer/OtlpMetricDescriptorTest.java

```diff
@@ -0,0 +1,87 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.function.Function;
+import java.util.function.IntFunction;
+
+import org.junit.Test;
+
+import io.aklivity.zilla.runtime.engine.config.KindConfig;
+import io.aklivity.zilla.runtime.engine.metrics.Metric;
+
+public class OtlpMetricDescriptorTest
+{
+    @Test
+    public void shouldDescribeServerMetric()
+    {
+        // GIVEN
+        Metric metric = mock(Metric.class);
+        when(metric.name()).thenReturn("http.request.size");
+        when(metric.kind()).thenReturn(Metric.Kind.HISTOGRAM);
+        when(metric.unit()).thenReturn(Metric.Unit.BYTES);
+        when(metric.description()).thenReturn("description");
+        Function<String, Metric> metricResolver = mock(Function.class);
+        when(metricResolver.apply("http.request.size")).thenReturn(metric);
+        IntFunction<KindConfig> resolveKind = mock(IntFunction.class);
+        when(resolveKind.apply(42)).thenReturn(KindConfig.SERVER);
+        OtlpMetricsDescriptor descriptor = new OtlpMetricsDescriptor(metricResolver, resolveKind);
+
+        // WHEN
+        String kind = descriptor.kind("http.request.size");
+        String name = descriptor.nameByBinding("http.request.size", 42);
+        String description = descriptor.description("http.request.size");
+        String unit = descriptor.unit("http.request.size");
+
+        // THEN
+        assertThat(kind, equalTo("histogram"));
+        assertThat(name, equalTo("http.server.request.size"));
+        assertThat(description, equalTo("description"));
+        assertThat(unit, equalTo("bytes"));
+    }
+
+    @Test
+    public void shouldDescribeClientMetric()
+    {
+        // GIVEN
+        Metric metric = mock(Metric.class);
+        when(metric.name()).thenReturn("http.request.size");
+        when(metric.kind()).thenReturn(Metric.Kind.HISTOGRAM);
+        when(metric.unit()).thenReturn(Metric.Unit.BYTES);
+        when(metric.description()).thenReturn("description");
+        Function<String, Metric> metricResolver = mock(Function.class);
+        when(metricResolver.apply("http.request.size")).thenReturn(metric);
+        IntFunction<KindConfig> resolveKind = mock(IntFunction.class);
+        when(resolveKind.apply(42)).thenReturn(KindConfig.CLIENT);
+        OtlpMetricsDescriptor descriptor = new OtlpMetricsDescriptor(metricResolver, resolveKind);
+
+        // WHEN
+        String kind = descriptor.kind("http.request.size");
+        String name = descriptor.nameByBinding("http.request.size", 42);
+        String description = descriptor.description("http.request.size");
+        String unit = descriptor.unit("http.request.size");
+
+        // THEN
+        assertThat(kind, equalTo("histogram"));
+        assertThat(name, equalTo("http.client.request.size"));
+        assertThat(description, equalTo("description"));
+        assertThat(unit, equalTo("bytes"));
+    }
+}
```

Comment:
This unit test feels too tight to the implementation so it won't remain stable under refactoring.

Prefer an IT that verifies the protocol output so that implementation can be refactored and verified against unmodified IT.

Note: with the IT in place, this unit test can then be removed as the behavior under test is already covered.